js-envs/src/main/scala/org/scalajs/jsenv/rhino/ScalaJSCoreLib.scala

```diff
@@ -105,8 +109,8 @@ private[rhino] class ScalaJSCoreLib(linkingUnit: LinkingUnit) {
   private def getSourceMapper(fileName: String, untilLine: Int) = {
     val linked = providers(fileName.stripSuffix(PseudoFileSuffix))
     val mapper = new Printers.ReverseSourceMapPrinter(untilLine)
-    val desugared =
-      new ScalaJSClassEmitter(ECMAScript51Global, linkingUnit).genClassDef(linked)
+    val classEmitter = incClassEmitter.newScalaJSClassEmitter(linkingUnit)
```

Comment:
It would probably be more efficient to create this `classEmitter` once and for all as a field of `ScalaJSCoreLib`.